golem-worker-executor-base/src/services/scheduler.rs

```diff
@@ -238,19 +293,39 @@ impl SchedulerServiceDefault {
                         // TODO: metrics
                     }
                 }
+                ScheduledAction::Invoke {
+                    owned_worker_id,
+                    idempotency_key,
+                    full_function_name,
+                    function_input,
+                } => {
+                    // TODO: We probably need more error handling here and retry the action when we fail to enqueue the invocation.
+                    // We don't really care that it completes here, but it needs to be persisted in the invocation queue.
+                    let result = self
+                        .worker_access
+                        .activate_and_invoke(
+                            owned_worker_id.clone(),
+                            idempotency_key,
+                            full_function_name.clone(),
+                            function_input,
+                        )
+                        .await;
+
+                    if let Err(e) = result {
+                        error!(
+                            worker_id = owned_worker_id.to_string(),
+                            full_function_name = full_function_name,
+                            "Failed to invoke worker with scheduled invocation: {e}"
+                        );
+                    };
+                }
             }
-        }
 
-        for owned_worker_id in owned_worker_ids {
-            let span = span!(
-                Level::INFO,
-                "scheduler",
-                worker_id = owned_worker_id.worker_id.to_string()
-            );
-            self.worker_access
-                .activate_worker(&owned_worker_id)
-                .instrument(span)
-                .await;
+            // We are completely done with the action, purge it from the queue
+            self.key_value_storage
```

Comment:
Moved this to the end so we are only removing scheduled actions if we actually successfully ran them.
Before we might have lost actions in the presence of failures